### ISSUE TITLE
Declaring bash functions using the `function` keyword, with function body in parens

### ISSUE THREAD

**Issue by DrHyde** [koalaman/shellcheck#2651]
#### For bugs
- Rule Id (if any, e.g. SC1000): 

SC1073, SC1065, SC1064, SC1072

- My shellcheck version (`shellcheck --version` or "online"): 

0.9.0

- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
function foo ( pwd )

```

#### Here's what shellcheck currently says:

```
$ shellcheck foo

In foo line 2:
function foo ( pwd )
^-- SC1073 (error): Couldn't parse this function. Fix to allow more checks.
               ^-- SC1065 (error): Trying to declare parameters? Don't. Use () and refer to params as $1, $2..


In foo line 3:

^-- SC1064 (error): Expected a { to open the function definition.
^-- SC1072 (error):  Fix any mentioned problems and try again.

For more information:
  https://www.shellcheck.net/wiki/SC1064 -- Expected a { to open the function...
  https://www.shellcheck.net/wiki/SC1065 -- Trying to declare parameters? Don...
  https://www.shellcheck.net/wiki/SC1072 --  Fix any mentioned problems and t...
```

#### Here's what I wanted or expected to see:

Either nothing, or if shellcheck strongly prefers that the function be defined as `foo() ( pwd )` (it doesn't complain about that at all) it should say so.

Having the function body in parens instead of braces is kosher, it just means that the function body is executed in a sub-shell.
